1-Draft/RFCNNNN-Multi-Line-Continuation.md

```diff
@@ -0,0 +1,301 @@
+could be used to enter multi-line parameter/argument parsing mode, and others
+were presented in the discussion however none of the other sigils that were
+presented could be used without breaking changes. When considering an alternate
+sigil, it must be something that can be identified as a unique token without
+breaking commands that accept multiple strings as positional parameters, such
+as Write-Host (which can write many sigils to the console) or commands external
+to PowerShell.
+
+### Enclosures instead of a sigil
+
+Instead of only requiring a single leading sigil, some users prefer the notion
+of enclosures such that the multi-line command parsing mode would have a very
+clear and well defined start and end. To meet that need, we could follow the
+here-string syntax in PowerShell as an example, offering syntax like the
+following:
+
+```PowerShell
+. {"./plink.exe" @`
+    --% 
+    $Hostname
+    -l $Username
+    -pw $Password
+    $Command
+`@}
+```
+
+All this would do is prevent newline tokens within the enclosures from being
+treated as statement terminators in the current statement.
+
+The closing closure could also be a recognized statement terminator even when
+used after the stop parsing sigil, allowing those commands to be wrapped across
+multiple lines as well.
+
+Like here-string enclosures, we could require that the opening closure be the
+last token on a line. Unlike here-string enclosures, however, it would be
+preferable if the closing closure did not have to be at the start of a line,
+since there is no need for it to be. It would simply have to be the first token
+on line to close the statement, allowing for indentation within scripts.
+
+This alternative also allows for blank lines to be used within the enclosures,
+such that Scenario 3 in @dragonwolf83's comment could be supported and written
+like this:
+
+```PowerShell
+New-ADUser @`
+    -Name 'Jack Robinson'
+    -GivenName 'Jack'
+    -Surname 'Robinson'
+    -SamAccountName 'J.Robinson'
+
+    -UserPrincipalName (
+        'J.Robinson@enterprise.com'
+    )
+
+   # Get the list of regions for where a user would reside in to put the user into the correct region
+    -Path (
+        $region = Get-Region -Name 'Jack Robinson'
+        "OU=Users,OU=$region,DC=enterprise,DC=com"
+    )
+
+    -AccountPassword (
+        Read-Host -AsSecureString 'Input Password'
+    )
+
+    -Enabled $true
+`@
+
+Get-ChildItem @`
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+`@
+```
+
+The best part is that it doesn't appear this syntax would introduce a breaking
+change at all.
+
+If people feel the backtick is still not visible enough here (and therefore not
+desirable for this purpose), we should keep the `@` portion of the enclosures
+so that we can avoid breaking changes, and simply replace the backtick with
+something else. For example, we could do this instead:
+
+```PowerShell
+Get-ChildItem @-
```

Comment:
We don't believe this is a breaking change, but later feedback still informs why we don't feel like the new enclosure operator is a good idea.

Reply:
My preference is for the syntax without the enclosure operator (a single trailing `@`, with the statement terminator being a blank line or some other statement terminator such as a semi-colon). Regardless, in my experimentation this is not a breaking change either because the parser identifies `@` as an unrecognized token in either case.